code/src/context/DrawerContext.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  PropsWithChildren,
+  ReactElement,
+  createContext,
+  useContext,
+  useState
+} from "react";
+
+import Drawer from "@mui/material/Drawer";
+
+type DrawerContextType = {
+  openDrawer: (component: ReactElement) => void;
+  toggleDrawer: (component: ReactElement) => void;
+  closeDrawer: () => void;
+};
+
+type DrawerProviderProps = Required<PropsWithChildren>;
+
+const DrawerContext = createContext<DrawerContextType | null>(null);
+
+const DrawerProvider = ({ children }: DrawerProviderProps) => {
+  const [drawer, setDrawer] = useState<ReactElement | null>(null);
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const openDrawer = (component: ReactElement) => {
+    setIsOpen(true);
+    setDrawer(component);
+  };
+
+  const closeDrawer = () => {
+    setIsOpen(false);
+  };
+
+  const toggleDrawer = (component: ReactElement) => {
```

Comment:
maybe we can awoid this if/else ?

`drawer ? closeDrawer() : openDrawer(component);`